### ISSUE TITLE
Add concrete "hello world" example to App Developer getting started guide

### ISSUE THREAD

**Issue by adambkaplan** [devfile/api#1057]
## Which area this feature is related to?

/area documentation

## Which area is this documentation change is related to?

/area landing-page

## Issue details

The "Getting Started" guide for application developers does not contain any concrete examples on how to incorporate a devfile into your source code, nor does it provide guidance on how to choose the right one for you.

After reading the getting started page, I should be able to add a devfile to my source code repository and use it with a tool that supports devfiles (ex - `odo`).

Page: https://devfile.io/docs/2.2.0/application-developer

**Target Date: 05-30-2023**

**Comment by michael-valdron** [comment]
Thank you for your feedback @adambkaplan, one of our goals for Q2 2023 is to provide improvements to our doc pages with the new user in mind. I have opened an epic issue to track any other issues towards this goal: https://github.com/devfile/api/issues/1080

For this issue, we will specifically address the main subject of creating a straightforward "Hello world" example guide for getting started with devfiles which will also make reference to the relevant material for two of the devfile supporting developer tools `odo` and Eclipse Che so that a new user can try out their first devfile for themselves.

As for the *Application Developer* page, I have create another issue which will address this piece: https://github.com/devfile/api/issues/1079

**Comment by elsony** [comment]
Additional requirement:
This "hello world" guide will need to be referenced by:
1. the page https://devfile.io/docs/2.2.0/developing-with-devfiles as a natural progression for developing devfile after the Intro section
2. the pages https://devfile.io/docs/2.2.0/application-developer and https://devfile.io/docs/2.2.0/enterprise-architect-and-runtime-provider  as those are tasks that are likely to be done by those personas.

**Comment by michael-valdron** [comment]
Creating an instruction on creating a simple devfile for using the [expressjs hello world example](https://expressjs.com/en/starter/hello-world.html) application with `odo` v3. This instruction will go step by step on adding the fields needed to make this work to help a new adopter learn what each piece does. This generally flows the following way:
1. Create application source files
2. Create the devfile
3. Step by step construction of the devfile
4. How to run with `odo`
5. A extra piece to briefly show that `odo init` could be used instead to quickstart with the `nodejs` stack from the devfile registry

Also looking into an upcoming mirrored instruction with Eclipse Che.

**Comment by michael-valdron** [comment]
A draft for the hello world example with odo guide is contained in https://github.com/devfile/devfile-web/pull/87. The Eclipse Che version of this guide will need additional time for completion due to set backs experienced in other issues I am working on. Therefore, the upcoming sprint will become the new target.

**Comment by michael-valdron** [comment]
Final drafts of the hello world starter guides for odo version 3 and Eclipse Che are now ready for review: https://github.com/devfile/devfile-web/pull/87

**Comment by michael-valdron** [comment]
> Final drafts of the hello world starter guides for odo version 3 and Eclipse Che are now ready for review: [devfile/devfile-web#87](https://github.com/devfile/devfile-web/pull/87)

Adjusted target date for this issue to give the review process more time. Reaching out to members of odo and Eclipse Che to provide reviews on the respective guides.

**Comment by michael-valdron** [comment]
PR is waiting on additional reviews: https://github.com/devfile/devfile-web/pull/87#issuecomment-1533703483

**Comment by michael-valdron** [comment]
> PR is waiting on additional reviews: [devfile/devfile-web#87 (comment)](https://github.com/devfile/devfile-web/pull/87#issuecomment-1533703483)

Received reviews from odo team today, provided changes and followups to feedback given.

**Comment by michael-valdron** [comment]
> > PR is waiting on additional reviews: [devfile/devfile-web#87 (comment)](https://github.com/devfile/devfile-web/pull/87#issuecomment-1533703483)
> 
> Received reviews from odo team today, provided changes and followups to feedback given.

Extending the target date to provide more time on the reviews.

**Comment by michael-valdron** [comment]
> https://github.com/devfile/devfile-web/pull/87#pullrequestreview-1417167301

Will need to push another draft for the Eclipse Che guide as Red Hat OpenShift Dev Spaces, which I was using in creating this guide, does not use the latest stable version of Che.

To ensure the information is consistent and easy for a new user, I will use the same version of Eclipse Che as what is in use for Red Hat OpenShift Dev Spaces in Red Hat Developer Sandbox to revise the guide for devfile `2.1.0`.

To cover the up to date Eclipse Che which uses the devfile `2.2.0` spec, I will create a mirror of the current guide under the `2.2.0` section of the docs with the up to date information on writing the hello world devfile using `2.2.0` spec in Eclipse Che. In addition, I will also provide a reference to the `2.1.0` version of the guide for setting up the devfile in Red Hat Developer Sandbox until a newer version of Red Hat Dev Spaces with `2.2.0` support is available.

**Comment by michael-valdron** [comment]
No longer a blocker for #1113 and #1114.

**Comment by michael-valdron** [comment]
> To ensure the information is consistent and easy for a new user, I will use the same version of Eclipse Che as what is in use for Red Hat OpenShift Dev Spaces in Red Hat Developer Sandbox to revise the guide for devfile `2.1.0`.
>
> To cover the up to date Eclipse Che which uses the devfile `2.2.0` spec, I will create a mirror of the current guide under the `2.2.0` section of the docs with the up to date information on writing the hello world devfile using `2.2.0` spec in Eclipse Che. In addition, I will also provide a reference to the `2.1.0` version of the guide for setting up the devfile in Red Hat Developer Sandbox until a newer version of Red Hat Dev Spaces with `2.2.0` support is available.

The `2.1.0` doc for Eclipse Che has been converted into a guide targeting **OpenShift Dev Spaces**, a new guide targeting Eclipse Che has been added to the `2.2.0` spec doc pages under WIP PR devfile/devfile-web#87.

**Comment by michael-valdron** [comment]
"Quick start with Eclipse Che" has been rewritten to let the reader create the devfile in the local workspace and reload into the workspace the devfile is complete. This should address the previous review comment: https://github.com/devfile/devfile-web/pull/87#discussion_r1187678110

Drafts for "Quick start with Eclipse Che" and "Quick start with OpenShift Dev Spaces" are ready for re-review: https://github.com/devfile/devfile-web/pull/87

**Comment by michael-valdron** [comment]
> "Quick start with Eclipse Che" has been rewritten to let the reader create the devfile in the local workspace and reload into the workspace the devfile is complete. This should address the previous review comment: [devfile/devfile-web#87 (comment)](https://github.com/devfile/devfile-web/pull/87#discussion_r1187678110)
> 
> Drafts for "Quick start with Eclipse Che" and "Quick start with OpenShift Dev Spaces" are ready for re-review: [devfile/devfile-web#87](https://github.com/devfile/devfile-web/pull/87)

Waiting on review for this PR, will need to extend into next sprint to give this additional time.

**Comment by michael-valdron** [comment]
During a review discussion, https://github.com/devfile/devfile-web/pull/87#discussion_r1201348280, it is decided that the version of the draft guide for 'OpenShift Dev Spaces' is to be removed and references to it under "Quick start with Eclipse Che" are replaced with a guide on Eclipse Che documentation called '[Eclipse Che hosted by Red Hat](https://www.eclipse.org/che/docs/stable/hosted-che/hosted-che/)'.

**Comment by michael-valdron** [comment]
"Quick start with Eclipse Che" will now be available under all devfile versions: https://github.com/devfile/devfile-web/pull/87#discussion_r1201217667

**Comment by michael-valdron** [comment]
> Additional requirement: This "hello world" guide will need to be referenced by:
> 
>  1. the page https://devfile.io/docs/2.2.0/developing-with-devfiles as a natural progression for developing devfile after the Intro section
> 
> 2. the pages https://devfile.io/docs/2.2.0/application-developer and https://devfile.io/docs/2.2.0/enterprise-architect-and-runtime-provider  as those are tasks that are likely to be done by those personas.

Added references to these pages, waiting for a review on these: https://github.com/devfile/devfile-web/pull/87

**Comment by michael-valdron** [comment]
https://github.com/devfile/devfile-web/pull/87 has received approval from members of devfile, odo, and che teams and will be merging shortly.

Thank you everyone for providing your reviews and input!

cc: @adambkaplan @elsony @kim-tsao @mike-hoang @feloy @kadel @amisevsk @l0rd
